packages/js-runtime/src/runtime/http/URL.ts

```diff
@@ -80,6 +81,48 @@
       throw new Error('Not implemented');
     }
 
+    get href(): string {
+      // TODO: username and password
+      let href = this.protocol + '//' + this.host + this.pathname + this.search + this.hash;
```

Comment:
Can we also implement username/password in this PR? We should only add the username -> password separator `:` if both of them are non-empty strings. If at least the username is set, we can add a `@` separator before the host.